testing/unittests/espapicmdtests.cpp

```diff
@@ -312,7 +312,14 @@ void sortAndWriteBack(string& input)
     }
 }
 
-Init setup;
+Init& ensureSetup()
+{
+    static std::unique_ptr<Init> globalSetup;
```

Comment:
If you are coding something like this then the best way would be

```
Init& ensureSetup()
{
    static std::unique_ptr<Init> globalSetup(new Init());
    return *globalSetup.get();
}

```
or

```
Init& ensureSetup()
{
    static Init globalSetup;
    return &globalSetup;
}

```
They are both guaranteed to be thread safe (and the compiler takes care of the details).  However even better is to make it a member.

statics in functions are best avoided because of the inefficiency introduced by this thread safety....